TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -585,6 +591,54 @@ private void WritePsmResults()
             }
         }
 
+        private void UpdateSpectralLibrary()
```

Comment:
May want to consider taking these functions out of the PostSearchAnalysisTask and moving them into one of the SpectralLibrary classes. Not mandatory, but might make things more organized.

Reply:
Thanks for your advice! I put the library update function here as the function of spectral library generation is here. They are the same kind of functions.